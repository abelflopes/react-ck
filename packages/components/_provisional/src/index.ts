export * from "./image";

export * from "./avatar";

export * from "./dropdown";

export * from "./dock";

export * from "./breadcrumbs";

export * from "./scrollable-container";

export * from "./position-engine";

export * from "./top-bar";

export * from "./snackbar";

export * from "./overlay-spinner";

export * from "./menu";

export * from "./tooltip";

export * from "./popover";

export * from "./chat";

export * from "./file-uploader";

export * from "./attachment";

export * from "./snippet";
