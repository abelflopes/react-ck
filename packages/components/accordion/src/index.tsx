import React from "react";
import { List, type ListProps } from "@react-ck/list";
import { Collapse } from "@react-ck/collapse";
import styles from "./styles/index.module.scss";

export interface AccordionItem {
  header: NonNullable<React.ReactNode>;
  children: NonNullable<React.ReactNode>;
}

export interface AccordionProps extends Omit<ListProps, "items"> {
  items: AccordionItem[];
}

/**
 * An accordion is a vertically stacked list of interactive items. Each item can be "collapsed" with only a summary visible or “expanded” to show the full content for that item.
 * @param props - {@link AccordionProps}
 * @returns a React element
 */

export const Accordion = ({
  items,
  ...otherProps
}: Readonly<AccordionProps>): React.ReactElement => (
  <List
    {...otherProps}
    items={items.map(({ header, children }) => (
      <Collapse key={String(JSON.stringify(header))} header={header}>
        <div className={styles.content}>{children}</div>
      </Collapse>
    ))}
  />
);
