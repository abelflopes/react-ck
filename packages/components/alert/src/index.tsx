import React from "react";
import classNames from "classnames";
import styles from "./styles/index.module.scss";
import { Text } from "@react-ck/text";

export interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
  heading?: string;
  /** Specifies the visual style of the alert  */
  skin?: "neutral" | "primary" | "negative" | "average" | "positive";
}

// TODO: add dismiss

/**
 * Alert is a short message that provides contextual feedback in a prominent way.
 * @param props - {@link AlertProps}
 * @returns a React element
 */

export const Alert = ({
  heading,
  skin = "neutral",
  children,
  className,
  ...otherProps
}: Readonly<AlertProps>): React.ReactElement => (
  <div {...otherProps} className={classNames(styles.root, styles[skin], className)}>
    {heading && (
      <Text type="h4" as="p" variation="bold" margin={false} className={styles.heading}>
        {heading}
      </Text>
    )}
    {children}
  </div>
);
