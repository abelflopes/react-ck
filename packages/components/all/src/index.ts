export * from "./text";

export * from "./divider";

export * from "./banner";

export * from "./card";

export * from "./container";

export * from "./chip";

export * from "./collapse";

export * from "./list";

export * from "./overlay";
