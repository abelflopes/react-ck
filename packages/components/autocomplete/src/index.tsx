import React, { useEffect, useId, useMemo } from "react";
import styles from "./styles/index.module.scss";
import classNames from "classnames";
import { useFormFieldContext, type FormFieldProps } from "@react-ck/form-field";
import { AutocompleteOption } from "./AutocompleteOption";

interface AutocompleteProps extends React.InputHTMLAttributes<HTMLInputElement> {
  skin?: FormFieldProps["skin"];
}

/**
 * Autocomplete is a text input that proactively suggests options based on the userâ€™s initial input. The suggested options appear alongside the input in a panel.
 * @param props - {@link AutocompleteProps}
 * @returns a React element
 */

const Autocomplete = ({
  skin,
  id,
  className,
  children,
  ...props
}: Readonly<AutocompleteProps>): React.ReactElement => {
  const listId = useId();
  const formFieldContext = useFormFieldContext();

  const computedSkin = useMemo(
    () => formFieldContext?.skin ?? skin ?? "default",
    [formFieldContext?.skin, skin],
  );

  const computedId = useMemo(() => formFieldContext?.id ?? id, [formFieldContext?.id, id]);

  // Validate usage inside form field
  useEffect(() => {
    // Is not inside form field, skip
    if (formFieldContext === undefined) return;

    // Is inside form field
    if (skin)
      throw new Error("When using autocomplete inside form field, define skin on the form field");
    else if (id)
      throw new Error("When using autocomplete inside form field, define id on the form field");
  }, [formFieldContext, id, skin]);

  return (
    <>
      <input
        {...props}
        id={computedId}
        list={listId}
        className={classNames(
          styles.root,
          formFieldContext === undefined && styles.standalone,
          className,
          styles[`skin_${computedSkin}`],
        )}
      />
      <datalist id={listId}>{children}</datalist>
    </>
  );
};

Autocomplete.Option = AutocompleteOption;

export { Autocomplete, type AutocompleteProps };
