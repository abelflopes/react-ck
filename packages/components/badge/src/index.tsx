import classNames from "classnames";
import * as styles from "./styles/index.module.scss";
import React from "react";

export interface BadgeProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Specifies the visual style of the badge  */
  skin?: "neutral" | "primary" | "negative" | "average" | "positive";
}

/**
 * Badge is a compact label that appears beside a primary interface area which is used to represent status or metadata for that area..
 * @param props - {@link BadgeProps}
 * @returns a React element
 */

export const Badge = ({
  skin = "neutral",
  className,
  ...otherProps
}: Readonly<BadgeProps>): React.ReactElement => (
  <span {...otherProps} className={classNames(styles.root, styles[skin], className)} />
);
