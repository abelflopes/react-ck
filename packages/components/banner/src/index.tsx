import React from "react";
import styles from "./styles/index.module.scss";
import classNames from "classnames";
import { Container } from "@react-ck/container";
import { ThemeContextProvider } from "@react-ck/theme";
import { Overlay } from "@react-ck/overlay";

export interface BannerProps extends React.HTMLAttributes<HTMLDivElement> {
  /** The source URL of the banner image */
  src: string;
  /** The alternative text for the banner image */
  alt?: string;
  /** An array of React nodes representing actions or buttons displayed on the banner */
  actions?: React.ReactNode[];
}

/**
 * Large that appears as one of the first items on a page.
 * Itâ€™s a primary visual element that is used to focus user attention on key content or promotions
 * @param props - {@link BannerProps}
 * @returns a React element
 */

export const Banner = ({
  src,
  alt,
  actions,
  children,
  className,
  ...otherProps
}: Readonly<BannerProps>): React.ReactElement => (
  <ThemeContextProvider
    value={{
      inverted: true,
    }}>
    <div className={classNames(styles.root, className)} {...otherProps}>
      <Container className={styles.container} variation="big" spacingY>
        {children}

        {actions?.length ? <div className={styles.actions}>{actions}</div> : null}
      </Container>

      <img className={styles.image} alt={alt} src={src} />

      <Overlay />
    </div>
  </ThemeContextProvider>
);
