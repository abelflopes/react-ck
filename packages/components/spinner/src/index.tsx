import classNames from "classnames";
import styles from "./styles/index.module.scss";
import React from "react";
import { Icon, type IconProps } from "@react-ck/icon";
import { IconSpinner } from "@react-ck/icon/icons/IconSpinner";

export interface SpinnerProps extends Omit<React.HTMLAttributes<HTMLSpanElement>, "children"> {
  size?: IconProps["size"];
  skin?: "text" | "primary";
}

/**
 * Spinner is a visual indicator that shows the status of an ongoing process happening in the background.
 * Itâ€™s often used to communicate that the user has to wait before the interface is ready for interaction.
 * @param props - {@link SpinnerProps}
 * @returns a React element
 */

export const Spinner = ({
  size = "l",
  skin = "primary",
  className,
  ...otherProps
}: Readonly<SpinnerProps>): React.ReactElement => (
  <span {...otherProps} className={classNames(styles.root, styles[skin], className)}>
    <Icon Icon={IconSpinner} size={size} />
  </span>
);
