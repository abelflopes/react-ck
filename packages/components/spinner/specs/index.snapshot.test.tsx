import React from "react";
import { Spinner } from "../src/index";
import renderer from "react-test-renderer";

describe("snapshot Spinner", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Spinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
