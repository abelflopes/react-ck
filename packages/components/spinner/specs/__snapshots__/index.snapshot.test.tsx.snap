// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Spinner renders correctly 1`] = `
<span
  className="root primary"
>
  <span
    className="VsR_m2r2hGIr7NfX85BU mBOmzNBv3utkRxhXCqIx sf93RNkSSnS5VkGCtSLh"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
      />
    </svg>
  </span>
</span>
`;
